internal/types/scope/scope.go

```diff
@@ -38,3 +44,24 @@ var Map = map[string]Type{
 	Org.String():     Org,
 	Project.String(): Project,
 }
+
+// AllowedIn returns the set of Scopes a known Resource type is allowed in.
+func AllowedIn(ctx context.Context, r resource.Type) ([]Type, error) {
+	const op = "scope.AllowedIn"
+	switch r {
+	case resource.Alias, resource.Billing, resource.Worker:
+		return []Type{Global}, nil
+	case resource.Account, resource.AuthMethod, resource.AuthToken, resource.ManagedGroup, resource.Policy, resource.Scope, resource.SessionRecording, resource.StorageBucket, resource.User:
+		return []Type{Global, Org}, nil
+	case resource.Group, resource.Role:
+		return []Type{Global, Org, Project}, nil
+	case resource.CredentialLibrary, resource.Credential, resource.CredentialStore, resource.HostCatalog, resource.HostSet, resource.Host, resource.Session, resource.Target:
+		return []Type{Project}, nil
+	case resource.Unknown:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "unknown resource type")
+	case resource.All:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "resource type '*' is not supported")
+	default:
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "invalid resource type")
```

Comment:
Nit: maybe include the value to ease debugging when this happens: 
```suggestion
		return nil, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("invalid resource type: resource.Type(%d)", r))
```

Reply:
Updated: ef5ae30f730fa43579755d5a863fb7a1d96ab41a

Would you suggest wrapping the value in `resource.Type()`?

Reply:
I included the type name in the error message to make it slightly easier for users, but I think this is fine.